src/stores/DetailsForm/Store.ts

```diff
@@ -295,14 +318,56 @@ export const getInitialState = (
             workflow === 'materialization_create';
 
         if (connectorId) {
-            const { setHydrationErrorsExist } = get();
+            const {
+                setDataPlaneOptions,
+                setHydrationError,
+                setHydrationErrorsExist,
+            } = get();
+
+            let dataPlaneOptions: DataPlaneOption[] | null =
+                dataPlaneFeatureFlag === 'show_option' ? [] : null;
+
+            if (dataPlaneOptions !== null) {
+                const dataPlaneResponse = await getDataPlaneOptions();
+
+                if (
+                    !dataPlaneResponse.error &&
+                    dataPlaneResponse.data &&
+                    dataPlaneResponse.data.length > 0
+                ) {
+                    dataPlaneOptions = dataPlaneResponse.data.map(
+                        ({ data_plane_name, id }) => {
+                            const scope = getDataPlaneScope(data_plane_name);
+
+                            const { cluster, prefix, provider, region } =
+                                parseDataPlaneName(data_plane_name, scope);
+
+                            return {
+                                dataPlaneName: {
+                                    cluster,
+                                    prefix,
+                                    provider,
+                                    region,
+                                    whole: data_plane_name,
+                                },
+                                id,
+                                scope,
+                            };
+                        }
+                    );
+
+                    setDataPlaneOptions(dataPlaneOptions);
+                } else {
+                    setHydrationError(
+                        dataPlaneResponse.error?.message ??
+                            'An error was encountered initializing the details form. If the issue persists, please contact support.'
+                    );
```

Comment:
Not a fan of how the error is handled when no data-planes options could be found, but this seemed like the lesser of two evils. The alternative approach would be to listen to both `hydrationErrorsExist` and `hydrationError` in the `EntityCreate` and `EntityEdit` components and display (internationalized) content if `hydrationErrorsExist` is `true` but `hydrationError` is undefined.